src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +390,50 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                OutStream.Write(value);
+            }
+            else
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(value.Length);
+                try
+                {
+                    value.CopyTo(buffer);
+                    Write(buffer);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                }
+            }
+        }
+
+        public virtual void Write(ReadOnlySpan<char> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                Span<byte> bytes = new byte[value.Length * 2];
+                _encoding.GetBytes(value, bytes);
+                OutStream.Write(bytes);
```

Comment:
A few things:
1) There's no guarantee that value.Length * 2 is sufficient space.  This either needs to compute the actual or worst-case bound, e.g. GetMaxByteCount(value.Length).
2) There's no guarantee that GetBytes filled all of bytes... this call is potentially writing out dummy data at the end of the span.
3) This is allocating a byte array.  If we need an array, it'd be better to use something that doesn't require per-operation allocation, e.g. if it's small enough (say, < 256 bytes) using stackalloc to get stack space, or else getting a temporary array from the array pool.  BinaryWriter also has a couple of scratch buffers it may have allocated, _buffer and _largeByteBuffer (or something like that), which could be used if they were allocated and sufficient in size.